src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -803,8 +803,9 @@ export class ConnectionWidget extends lifecycle.Disposable {
 				this._callbacks.onAzureTenantSelection(tenant.id);
 			}
 			else {
-				// This should ideally never ever happen!
-				this._logService.error(`onAzureTenantSelected : Tenant list not found as expected, missing tenant on index ${tenantIndex}`);
+				// This can happen when connection dialog is opened with Azure MFA selected, but not with profile data.
+				// This is not an error since tenant is not available in this case, tenants will be populated for the account selected in account dropdown.
+				this._logService.debug(`onAzureTenantSelected : Tenant list not found as expected, missing tenant on index ${tenantIndex}`);
```

Comment:
1. Could you instead just not log these messages at all when the situation is expected? And then still log the error when it's unexpected?
2. Or at the very least if that's difficult/not possible I'd suggest adding a note in the log message that this may be expected (with any additional information that may help tell someone reading the logs whether this should be happening like the currently "selected" tenant)

Reply:
Thanks for the push, I was able to handle nulls and need not change these from being logged.. :)